test/utils/create-libp2p-node.js

```diff
@@ -8,33 +8,46 @@ const KadDHT = require('libp2p-kad-dht')
 const waterfall = require('async/waterfall')
 const PeerInfo = require('peer-info')
 const PeerId = require('peer-id')
+const defaultsDeep = require('@nodeutils/defaults-deep')
 
 class Node extends libp2p {
-  constructor (peerInfo, options) {
-    options = options || {}
-
-    const modules = {
-      transport: [new TCP()],
-      connection: {
-        muxer: Multiplex,
-        crypto: SECIO
+  constructor (_options) {
+    const defaults = {
+      modules: {
+        transport: [
+          TCP
+        ],
+        streamMuxer: [
+          Multiplex
+        ],
+        connEncryption: [
+          SECIO
+        ],
+        dht: _options.DHT ? KadDHT : undefined
       },
-      DHT: options.DHT ? KadDHT : undefined
+      config: {
+        dht: {},
+        EXPERIMENTAL: {
+          dht: Boolean(_options.DHT)
+        }
+      }
     }
 
-    super(modules, peerInfo, null, options.DHT || {})
+    delete _options.DHT
```

Comment:
I don't think it's a huge problem in this particular case but we should avoid using the `delete` operator on arguments to functions, as it'll change the `options` object that was passed into the constructor